### ISSUE TITLE
Setting `set -o vi` to enable Vi keybindings in .bashrc, causes Eshell in Doom to crash

### ISSUE THREAD

**Issue by ReneFroger** [doomemacs/doomemacs#6861]
### I confirm that...

- [X] I have searched the [issue tracker](https://github.com/doomemacs/doomemacs/issues), [documentation](https://docs.doomemacs.org), [FAQ](https://docs.doomemacs.org/-/faq), [Discourse](https://discourse.doomemacs.org), and [Google](https://google.com), in case this issue has already been reported/resolved.

- [X] I have read "[How to Debug Issues](https://doomemacs.org/d/how2debug)", and will use it to provide as much information about this issue as possible.

- [X] The issue can be reproduced on the **latest** available commit of Doom.

- [X] The issue can be reproduced on a stable release of Emacs, such as 27 or 28. *(Doom does not support development builds like 29+ or any version ending in .50 or .9x)*

### Expected behavior

When I type in Eshell process `find-file` <kbd>Tab</kbd>, I expect to have a list of possible file candidates to complete.

### Current behavior

I start Eshell. I navigate to a directory, then I type `find-file`, <kbd>[Space]</kbd> then doing <kbd>[tab]</kbd>, the Doom Emacs hangs and become unresponsive. I have to kill the Emacs process and restart again.

### Steps to reproduce

In my `.bashrc` I have only two following lines:
```
#enable Vi keybindings in Bash
set -o vi
```

In an clean Doom installation with the  `.bashrc` above, I start Eshell. I navigate to a directory, then I type `find-file`, <kbd>[Space]</kbd> then doing <kbd>[tab]</kbd>, the Doom Emacs hangs and become unresponsive. I have to kill the Emacs process and restart again.

When I comment out `set -o vi` in my `.bashrc`, then restart Emacs and load  with clean Doom configuration. 
Start Eshell, then I type `find-file`, <kbd>[Space]</kbd> then doing <kbd>[tab]</kbd>, then I get suggestions to autocomplete.

So the culprit is the line `set -o vi` that somehow causes Eshell only in Emacs Doom distribution to be unresponsive when calling for file name completion (not directory completion).

For some reason, I couldn't paste my system information. So a screenshot of my system and versions:

![image](https://user-images.githubusercontent.com/5211845/193106626-af2cdec0-1bee-4f1a-8888-f00d27b14cfe.png)

### System Information

https://pastebin.com/

**Comment by erik-overdahl** [comment]
If you set `vi` mode in `~/.inputrc` instead of `~/.bashrc`, do you get the same behavior?
https://wiki.archlinux.org/title/readline#Editing_mode

**Comment by ReneFroger** [comment]
No, after removing `set -o vi` from `.bashrc` and setting `vi` mode in `.inputrc`, I still get Vi keybindings in Bash, and Doom doesn't crash anymore. 

Basically, my issue is solved via this way. However, I believe settings in `.bashrc` shouldn't affect Doom's Eshell at anything. So I 'll leave this issue open because more users might be affected by this behavior in the future.
